doozer/doozerlib/util.py

```diff
@@ -687,23 +641,13 @@ async def run_oc(auth_file: Optional[str]):
         _, out, _ = await exectools.cmd_gather_async(cmd)
         return json.loads(out)
 
-    # If neither a username nor a password were provided, assume an auth file is being used or no auth at all
-    if not registry_username and not registry_password:
-        return await run_oc(auth_file=registry_config)
-
-    # Store the basic auth info in a temporary temp file to be used with --registry-config
-    with tempfile.NamedTemporaryFile(mode='w', prefix="_doozer_") as registry_config_file:
-        registry_config_file.write(get_registry_basic_auth(pullspec, registry_password, registry_username))
-        registry_config_file.flush()
-        return await run_oc(auth_file=registry_config_file.name)
+    return await run_oc(auth_file=registry_config)
```

Comment:
Same as above: can we remove the nested function `run_oc`, and the retry to the outer scope?